### PR TITLE
Adding projectOwnerId to Projects.Create

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description
[CLOUDP-95603: [Mongocli - Atlas] Add support for projectOwnerId flag with project creation/modification](https://jira.mongodb.org/browse/CLOUDP-95603)

Description:
Added query param [projectOwnerId](https://docs.atlas.mongodb.com/reference/api/project-create-one/#request-query-parameters) to `Projects.Create`

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/projects.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ type ProjectsService interface {
 	GetAllProjects(context.Context, *ListOptions) (*Projects, *Response, error)
 	GetOneProject(context.Context, string) (*Project, *Response, error)
 	GetOneProjectByName(context.Context, string) (*Project, *Response, error)
-	Create(context.Context, *Project) (*Project, *Response, error)
+	Create(context.Context, *Project, *ProjectOptions) (*Project, *Response, error)
```
Comment (gssbzn) [inline mongodbatlas/projects.go:45]: this is a breaking change, I'm a bit conflicted here, I'm ok to do this but yes, breaking, a non breaking alternative is to add a new method `CreateWithOptions` or something but let's discuss with DoU

File: mongodbatlas/projects.go [hunk 2]
```diff
@@ -99,6 +99,10 @@ type TeamsAssigned struct {
 	TotalCount int       `json:"totalCount"`
 }
 
+type ProjectOptions struct {
```
Comment (gssbzn) [inline mongodbatlas/projects.go:102]: [nit] let's make this create specific 
```suggestion
type CreateProjectOptions struct { // or any other similar name 
```

**Review by gssbzn — APPROVED** [review]
LGTM but let's wait for DoU

File: mongodbatlas/projects.go (hunk 1 above)
Comment (thetonymaster) [reply mongodbatlas/projects.go]: Let's go with the second option unless it's something we want to do with terraform too, then we can review tomorrow, today we are OOO

Comment (gssbzn) [reply mongodbatlas/projects.go]: this will need to happen to terraform eventually, this add the ability to specify the project owner instead of the current behaviour of picking oldest org member, this param is optional

File: mongodbatlas/projects.go [hunk 3]
```diff
@@ -177,12 +181,17 @@ func (s *ProjectsServiceOp) GetOneProjectByName(ctx context.Context, projectName
...
 //
 // See more: https://docs.atlas.mongodb.com/reference/api/project-create-one/
-func (s *ProjectsServiceOp) Create(ctx context.Context, createRequest *Project) (*Project, *Response, error) {
+func (s *ProjectsServiceOp) Create(ctx context.Context, createRequest *Project, opts *CreateProjectOptions) (*Project, *Response, error) {
 	if createRequest == nil {
 		return nil, nil, NewArgError("createRequest", "cannot be nil")
 	}
 
-	req, err := s.Client.NewRequest(ctx, http.MethodPost, projectBasePath, createRequest)
+	path, err := setListOptions(projectBasePath, opts)
```
Comment (thetonymaster) [inline mongodbatlas/projects.go:189]: does `setListOptions` checks if opt is nil?

File: mongodbatlas/projects_test.go [hunk 4]
```diff
@@ -219,7 +219,9 @@ func TestProject_Create(t *testing.T) {
 		}`)
 	})
 
-	project, _, err := client.Projects.Create(ctx, createRequest)
+	opts := &CreateProjectOptions{ProjectOwnerID: "1"}
```
Comment (thetonymaster) [inline mongodbatlas/projects_test.go:222]: Can we add a test with `CreateProjectOptions` set to nil, and check if the URI is formed correctly in both too?

**Review by thetonymaster — APPROVED** [review]

**Comment by thetonymaster** [comment]
Not blocking comments, just some suggestions

File: mongodbatlas/projects.go (hunk 1 above)
Comment (thetonymaster) [reply mongodbatlas/projects.go]: In  a second thought, don't I really find a case to have two functions for this since both will do pretty much the same thing, except for adding the arguments on the URI so it's ok, we will need to eventually update terraform too.

File: mongodbatlas/projects.go (hunk 3 above)
Comment (andreaangiolillo) [reply mongodbatlas/projects.go:189]: yes https://github.com/mongodb/go-client-mongodb-atlas/blob/a40fdd6c5807714665e121eff8144d942790ce49/mongodbatlas/mongodbatlas.go#L546

File: mongodbatlas/projects_test.go (hunk 4 above)
Comment (andreaangiolillo) [reply mongodbatlas/projects_test.go:222]: sure

Comment (andreaangiolillo) [reply mongodbatlas/projects_test.go:222]: done

**Comment by andreaangiolillo** [comment]
@gssbzn @thetonymaster ready for review

**Review by gssbzn — APPROVED** [review]
LGTM

**Comment by thetonymaster** [comment]
Already approved :)
